crates/fuel-core/src/graphql_api/service.rs

```diff
@@ -141,20 +142,24 @@ impl RunnableTask for Task {
     }
 }
 
+// Need a seperate Data Object for each Query endpoint, cannot be avoided
 pub fn new_service(
     config: Config,
-    database: Database,
     schema: CoreSchemaBuilder,
-    producer: BlockProducer,
+    query_data: Box<dyn QueryData>,
```

Comment:
We don't need to pass any `Box<dyn QueryData>`. The `Database` itself already should implement `QueryData` traits because of your blanket implementation.

And seems `QueryData` should be a super trait for `DatabasePort`.

Reply:
Removed QueryData